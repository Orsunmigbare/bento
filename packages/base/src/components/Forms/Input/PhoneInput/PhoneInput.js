import React from 'react'
import Input, { InputPropTypes } from '../Input'
import { phoneMasks } from '../masks'

const PhoneInput = ({mask, locale = 'nb', ...otherProps}) => (
  <Input type={'tel'} mask={phoneMasks[locale]} {...otherProps} />
)

PhoneInput.propTypes = InputPropTypes

export default PhoneInput