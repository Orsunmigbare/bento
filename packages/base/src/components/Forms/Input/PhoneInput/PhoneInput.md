```jsx harmony
<PhoneInput value={92284082} id="dlkn2"/>
```

```jsx harmony
<PhoneInput value={92284082} locale={'nb'} guide={false} id="2kjen" />
```

```jsx harmony
<PhoneInput value={92284082} locale={'nb'} label={'Telefon'} id="dnwqu" />
```
