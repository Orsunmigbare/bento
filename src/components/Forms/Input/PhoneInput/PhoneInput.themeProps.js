import { themePropTypes } from "../../../../theming"

const themeProps = {
  wrapper: {
    name: "PHONE_WRAPPER",
    description: "Wrapper style",
    type: themePropTypes.style
  },
  countryCode: {
    name: "PHONE_COUNTRY_CODE",
    description: "Country code style",
    type: themePropTypes.style
  },
  input: {
    name: "PHONE_INPUT",
    description: "Specific Input style",
    type: themePropTypes.style
  }
}

export default themeProps
