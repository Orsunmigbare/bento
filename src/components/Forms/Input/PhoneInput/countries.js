import { countries } from "libphonenumber-js/metadata.min.json"
import { distinct } from "../../../../utils/distinct"
export const NORWAY = {
  name: "NO",
  code: "47"
}

export default distinct(
  Object.keys(countries)
    .map(name => {
      return {
        code: countries[name][0],
        name
      }
    })
    .sort((a, b) => {
      return parseInt(a.code, 10) < parseInt(b.code, 10)
    }),
  "code"
)
