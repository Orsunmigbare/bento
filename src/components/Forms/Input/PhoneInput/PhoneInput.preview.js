import React from "react"
import PhoneInput from "./PhoneInput"
import Toggle from "../../Toggle/Toggle"
import Label from "../../Label/Label"

const preview = {
  title: "PhoneInput",
  category: "Components/Forms/Input",
  component: PhoneInput,
  render: props => (
    <PhoneInput label={"PhoneInput"} id="dsf4dfs456" {...props} />
  ),
  renderExampleController: ({ setComponentState }) => (
    <React.Fragment>
      <Label for="guidePhoneInput">Toggle guides</Label>
      <Toggle
        id="guidePhoneInput"
        onChange={e => setComponentState({ guide: e.target.checked })}
      >
        Guide
      </Toggle>
    </React.Fragment>
  )
}

export default preview
