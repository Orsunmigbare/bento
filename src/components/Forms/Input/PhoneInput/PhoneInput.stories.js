import React from "react"
import { storiesOf } from "@storybook/react"
import PhoneInput from "./PhoneInput"

storiesOf("components|base/Forms/Input/PhoneInput", module).add(
  "no props",
  () => <PhoneInput label={"PhoneInput"} id="9852145" />
)
