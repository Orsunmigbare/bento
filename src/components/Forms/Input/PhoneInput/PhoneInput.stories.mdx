import { Meta, Story, Preview, Props } from '@storybook/addon-docs/blocks';
import PhoneInput from "./PhoneInput"

<Meta title="Components|Forms/Input/PhoneInput" component={PhoneInput}/>


# PhoneInput
This components controlls user input so that they can only enter numbers with the format of a PhoneInput

```javascript
import {PhoneInput} from "@staccx/bento"
```

## Standard PhoneInput
The inputfield only allowed numbers and formats the input 3-2-3 and max input of 8 digits

<Preview>
  <Story name="Standard">
   <PhoneInput id="9852145" />
  </Story>
</Preview>


## With Country code
Also allowes selecting contry code from an array



## Props for PhoneInput
<Props of={PhoneInput} />
