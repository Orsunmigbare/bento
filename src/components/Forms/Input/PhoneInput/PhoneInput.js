import React from "react"
import PropTypes from "prop-types"
import styled from "styled-components"
import Input, { InputPropTypes } from "../Input"
import "cleave.js/dist/addons/cleave-phone.no.js"
import { applyVariants, spacing } from "../../../../theming"
import countries, { NORWAY } from "./countries"
import SelectSimple from "../../SelectSimple/SelectSimple"
import Label from "../../Label/Label"
import themeProps from "./PhoneInput.themeProps"
import { componentCreateFactory } from "../../../../theming/utils/createVariantsFunctionFactory"

/**
 * Input for Phone-numbers.
 * TODO: normalize country code and locale
 */

const PhoneInput = React.forwardRef(
  (
    {
      locale,
      label,
      showCountryCode = false,
      initialCountry = NORWAY,
      initialCountryCode = NORWAY.code,
      onCountryChange,
      ...otherProps
    },
    ref
  ) => {
    return (
      <PhoneInputWrapper {...otherProps}>
        <Label htmlFor="test" {...otherProps}>
          {label}
        </Label>
        <Wrap {...otherProps}>
          {showCountryCode && (
            <CountryCode>
              <SelectSimple
                {...otherProps}
                items={countries}
                itemToString={item => `+${item.code}`}
                itemToKey={item => item.code}
                initialSelectedItem={
                  initialCountryCode
                    ? countries.find(c => c.code === initialCountryCode)
                    : initialCountry
                }
                onChange={country => {
                  if (onCountryChange) {
                    onCountryChange(country)
                  }
                }}
              >
                {countries.map(country => (
                  <option key={country.code}>{country.code}</option>
                ))}
              </SelectSimple>
            </CountryCode>
          )}
          <PInput
            id="test"
            type="tel"
            ref={ref}
            options={{
              phone: true,
              phoneRegionCode: locale
            }}
            {...otherProps}
          />
        </Wrap>
      </PhoneInputWrapper>
    )
  }
)

const PhoneInputWrapper = styled.div`
  position: relative;
`

const Wrap = styled.div`
  width: 100%;
  position: relative;
  display: flex;
  align-items: flex-end;
  justify-content: space-between;
  ${applyVariants(themeProps.wrapper)};
`

const CountryCode = styled.div`
  flex-grow: 0.3;
  margin-right: ${spacing.small};
`

const PInput = styled(Input)`
  flex-grow: 1;
  ${applyVariants(themeProps.input)};
`

PhoneInput.propTypes = {
  ...InputPropTypes,
  locale: PropTypes.oneOf(["no"]),
  onChange: PropTypes.func,
  guide: PropTypes.bool,
  id: PropTypes.string.isRequired
}

PhoneInput.defaultProps = {
  locale: "no",
  guide: false
}
PhoneInput.themeProps = themeProps
PhoneInput.createVariants = componentCreateFactory(PhoneInput)

export default PhoneInput
