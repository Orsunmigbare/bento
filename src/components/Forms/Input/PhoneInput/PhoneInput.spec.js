import React from "react"
import renderer from "react-test-renderer"
import { ThemeProvider } from "styled-components"
import baseTheme from "../../../../theming/themes/baseTheme"
import PhoneInput from "./PhoneInput"

describe("PhoneInput", () => {
  describe("Snapshots", () => {
    it("ID, children and label", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <PhoneInput id="test" label="test">
              <p>test</p>
            </PhoneInput>
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
  })
})
