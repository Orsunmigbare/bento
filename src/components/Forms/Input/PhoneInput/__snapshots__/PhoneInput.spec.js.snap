// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhoneInput Snapshots ID, children and label 1`] = `
<div
  className="sc-fzpans jRCOFi"
  id="test"
>
  <label
    className="sc-AxjAm hjwNwx"
    id="test"
  >
    test
  </label>
  <div
    className="sc-fzplWN hkvTOA"
    id="test"
  >
    <div
      className="sc-AxheI jmeqRy sc-fznKkj fyCGGL"
    >
      
      
      <input
        autoFocus={null}
        className="sc-Axmtr bDDSAq"
        disabled={false}
        hidden={false}
        id="test"
        name={null}
        onBlur={null}
        onChange={null}
        onKeyDown={null}
        placeholder={null}
        style={null}
        type="tel"
      >
        <p>
          test
        </p>
      </input>
    </div>
  </div>
</div>
`;
